tests/unittests/executor/test_executor.py

```diff
@@ -0,0 +1,157 @@
+import sys
+import time
+from multiprocessing import TimeoutError
+
+import pytest
+
+from orion.executor.base import AsyncException
+from orion.executor.dask_backend import Dask
+from orion.executor.multiprocess_backend import PoolExecutor
+from orion.executor.single_backend import SingleExecutor
+
+
+def multiprocess(n):
+    return PoolExecutor(n, "multiprocess")
+
+
+def thread(n):
+    return PoolExecutor(n, "threading")
+
+
+backends = [thread, multiprocess, Dask, SingleExecutor]
+
+
+def function(a, b, c):
+    return a + b * c
+
+
+def slow_function(a, b, c):
+    time.sleep(5)
+    return function(a, b, c)
+
+
+class BadException(Exception):
+    pass
+
+
+def bad_function(a, b, c):
+    raise BadException()
+
+
+@pytest.mark.parametrize("backend", backends)
+def test_execute_function(backend):
+    with backend(5) as executor:
+        future = executor.submit(function, 1, 2, c=3)
+        assert executor.wait([future]) == [7]
+
+
+@pytest.mark.parametrize("backend", backends)
+def test_execute_bad_function(backend):
+    with backend(5) as executor:
+        future = executor.submit(bad_function, 1, 2, 3)
+        with pytest.raises(BadException):
+            executor.wait([future])
+
+
+@pytest.mark.parametrize("backend", backends)
+def test_execute_async(backend):
+    with backend(5) as executor:
+        futures = [executor.submit(function, 1, 2, i) for i in range(10)]
+
+        total_task = len(futures)
+        results = executor.async_get(futures, timeout=1)
+
+        assert len(results) <= total_task, "Maybe not all tasks were completed"
+        assert len(results) > 0, "We got some results"
+        assert len(futures) == total_task - len(results), "Finished futures got removed"
```

Comment:
Looks like these 2 lines are testing the same thing. I think last one is the clearest one.

Reply:
removed